File: python/08_047.py
class ListNode:

    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

def partition_list(head, pivot):
    if not head:
        return None
        
    # Create dummy heads for both partitions
    less_head = ListNode(0)
    greater_head = ListNode(0)
    
    less = less_head
    greater = greater_head
    
    # Traverse the original list
    current = head
    while current:
        if current.val < pivot:
            less.next = current
            less = less.next
        else:
            greater.next = current
            greater = greater.next
        
        current = current.next
    
    # Connect the two lists
    greater.next = None
    less.next = greater_head.next
    
    return less_head.next

File: data structures and algorithms/cs/08_047.cs
public class ListNode {
    public int Val { get; set; }
    public ListNode Next { get; set; }

    public ListNode(int val) {
        Val = val;
    }
}

public class Solution {
    public ListNode PartitionList(ListNode head, int pivot) {
        if (head == null) {
            return null;
        }

        ListNode lessHead = new ListNode(0);
        ListNode greaterHead = new ListNode(0);
        ListNode less = lessHead;
        ListNode greater = greaterHead;
        ListNode current = head;

        while (current != null) {
            if (current.Val < pivot) {
                less.Next = current;
                less = less.Next;
            } else {
                greater.Next = current;
                greater = greater.Next;
            }
            current = current.Next;
        }

        greater.Next = null;
        less.Next = greaterHead.Next;
        return lessHead.Next;
    }
}